Add azurerm_api_management_backend to existing APIM resource via Terraform CLI fails with Plugin did not respond

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

Terraform v1.1.2
AzureRM v2.90.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_api_management_backend`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = ">=2.90.0"
    }
  }
  cloud {
    # The name of your Terraform Cloud organization.
    organization = "***"

    # The name of the Terraform Cloud workspace to store Terraform state files in.
    workspaces {
      name = "api-dev-test1"
    }
  }
}

# Configure Microsoft Azure Provider
provider "azurerm" {
  features {}

  subscription_id             = "***"
  client_id                   = "***"
  tenant_id                   = "***"
  client_certificate_path     = "***"
  client_certificate_password = var.client_certificate_password
}

# Resource Group Data
data "azurerm_resource_group" "rg" {
  name = "rg-dev"
}

# Existing APIM Resource Data
data "azurerm_api_management" "apim_dev" {
  name                = "apim-dev"
  resource_group_name = data.azurerm_resource_group.rg.name
}

resource "azurerm_api_management_backend" "back_end_slot" {
  name                = "func-api-dev-test1"
  resource_group_name = data.azurerm_resource_group.rg.name
  api_management_name = data.azurerm_api_management.apim_dev.name
  protocol            = "http"
  url                 = "https://func-api-dev-test1.azurewebsites.net/api"

  credentials {
    certificate = []
    header      = {}
    query       = {}
  }

  tls {
    validate_certificate_chain = false
    validate_certificate_name  = false
  }
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->
[Terraform Output](https://gist.github.com/jmburns/575f363e695c74b1f201dd87e0c0cd55)

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->
Included in Terraform Output above.

### Expected Behaviour

<!--- What should have happened? --->
Backend should have been successfully added to the APIM resource.

### Actual Behaviour

<!--- What actually happened? --->
Terraform apply failed with output above.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform init`
2. `terraform apply -var 'client_certificate_password="***"' -auto-approve

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->
Did not find any related GitHub issues.

Stack trace:

```
Stack trace from the terraform-provider-azurerm_v2.90.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 106 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement.expandApiManagementBackendCredentials({0xc000b293f0, 0x5dc86af, 0x6472c20})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/apimanagement/api_management_backend_resource.go:395 +0x405
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement.resourceApiManagementBackendCreateUpdate(0xc0010a2e00, {0x54f0ae0, 0xc00106b880})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/apimanagement/api_management_backend_resource.go:278 +0x5ec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0017a4420, {0x64bf8b8, 0xc0017746c0}, 0x2, {0x54f0ae0, 0xc00106b880})
```

--- a/internal/services/apimanagement/api_management_backend_resource.go
+++ b/internal/services/apimanagement/api_management_backend_resource.go
@@ -389,7 +389,7 @@ func resourceApiManagementBackendDelete(d *pluginsdk.ResourceData, meta interfac
 }
 
 func expandApiManagementBackendCredentials(input []interface{}) *apimanagement.BackendCredentialsContract {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
 	v := input[0].(map[string]interface{})

